# CS 4304 - Assignment 1

Dylan Ellsworth

201329133

September 20, 2021

### Question 1: Write that finds the number of unique fields of study

We import pandas and then open our students file, downloaded from D2L. I then decided to show some info on the dataframe, so it was easier to glance at for later questions.

In [1]:
import pandas as pd
df = pd.read_csv('students.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6221 entries, 0 to 6220
Data columns (total 9 columns):
 #   Column                                                      Non-Null Count  Dtype  
---  ------                                                      --------------  -----  
 0   REF_DATE                                                    6221 non-null   object 
 1   GEO                                                         6221 non-null   object 
 2   Institution type                                            6221 non-null   object 
 3   Registration status                                         6221 non-null   object 
 4   International Standard Classification of Education (ISCED)  6221 non-null   object 
 5   Field of study                                              6221 non-null   object 
 6   Gender                                                      6221 non-null   object 
 7   Status of student in Canada                                 6221 non-null   object 
 8  

In [2]:
df['Field of study'].nunique()

14

### Question 2: List all the unique degree types in the International Standard Classification of Education(ISCED) field.

An array showing the different degree types.

In [3]:
df['International Standard Classification of Education (ISCED)'].unique()

array(["Bachelor's or equivalent", "Master's or equivalent",
       'Doctoral or equivalent', 'Post-secondary non-tertiary education',
       'Short-cycle tertiary education',
       'Not applicable, International Standard Classification of Education (ISCED)',
       'Upper secondary education'], dtype=object)

### Question 3: Determine the number of entries in the data set that have institution type of University or College

We have 6221 students in either Univeristy or College, equal to the total number of rows. Therefore, every data entry is a student enrolled in either University or College.

In [4]:
df[df['Institution type'].str.contains('University')| df['Institution type'].str.contains('College')]

REF_DATE                        GEO Institution type  \
0     2018/2019  Newfoundland and Labrador       University   
1     2018/2019  Newfoundland and Labrador       University   
2     2018/2019  Newfoundland and Labrador       University   
3     2018/2019  Newfoundland and Labrador       University   
4     2018/2019  Newfoundland and Labrador       University   
...         ...                        ...              ...   
6216  2018/2019                Territories          College   
6217  2018/2019                Territories          College   
6218  2018/2019                Territories          College   
6219  2018/2019                Territories          College   
6220  2018/2019                Territories          College   

     Registration status  \
0      Full-time student   
1      Full-time student   
2      Full-time student   
3      Full-time student   
4      Full-time student   
...                  ...   
6216   Part-time student   
6217   Part-time student   
6218   Part-time student   
6219   Part-time student   
6220   Part-time student   

     International Standard Classification of Education (ISCED)  \
0                              Bachelor's or equivalent           
1                              Bachelor's or equivalent           
2                              Bachelor's or equivalent           
3                              Bachelor's or equivalent           
4                              Bachelor's or equivalent           
...                                                 ...           
6216  Not applicable, International Standard Classif...           
6217  Not applicable, International Standard Classif...           
6218  Not applicable, International Standard Classif...           
6219  Not applicable, International Standard Classif...           
6220  Not applicable, International Standard Classif...           

     Field of study          Gender  \
0     Education [1]     Male gender   
1     Education [1]     Male gender   
2     Education [1]   Female gender   
3     Education [1]   Female gender   
4     Education [1]  Gender unknown   
...             ...             ...   
6216   Unclassified     Male gender   
6217   Unclassified   Female gender   
6218   Unclassified   Female gender   
6219   Unclassified  Gender unknown   
6220   Unclassified  Gender unknown   

                    Status of student in Canada  VALUE  
0                             Canadian students  129.0  
1                        International students    3.0  
2                             Canadian students  243.0  
3                        International students    3.0  
4                             Canadian students    3.0  
...                                         ...    ...  
6216  Not reported, status of student in Canada  144.0  
6217                          Canadian students  810.0  
6218  Not reported, status of student in Canada  246.0  
6219                          Canadian students  216.0  
6220  Not reported, status of student in Canada   72.0  

[6221 rows x 9 columns]

###  Question 4: Compute the total number of students enrolled in Newfoundland and Labrador

There are 347 students enrolled in Newfoundland and Labrador

In [5]:
df[df['GEO']== 'Newfoundland and Labrador']

REF_DATE                        GEO Institution type  \
0    2018/2019  Newfoundland and Labrador       University   
1    2018/2019  Newfoundland and Labrador       University   
2    2018/2019  Newfoundland and Labrador       University   
3    2018/2019  Newfoundland and Labrador       University   
4    2018/2019  Newfoundland and Labrador       University   
..         ...                        ...              ...   
342  2018/2019  Newfoundland and Labrador          College   
343  2018/2019  Newfoundland and Labrador          College   
344  2018/2019  Newfoundland and Labrador          College   
345  2018/2019  Newfoundland and Labrador          College   
346  2018/2019  Newfoundland and Labrador          College   

    Registration status  \
0     Full-time student   
1     Full-time student   
2     Full-time student   
3     Full-time student   
4     Full-time student   
..                  ...   
342   Part-time student   
343   Part-time student   
344   Part-time student   
345   Part-time student   
346   Part-time student   

    International Standard Classification of Education (ISCED) Field of study  \
0                             Bachelor's or equivalent          Education [1]   
1                             Bachelor's or equivalent          Education [1]   
2                             Bachelor's or equivalent          Education [1]   
3                             Bachelor's or equivalent          Education [1]   
4                             Bachelor's or equivalent          Education [1]   
..                                                 ...                    ...   
342                     Short-cycle tertiary education           Unclassified   
343  Not applicable, International Standard Classif...           Unclassified   
344  Not applicable, International Standard Classif...           Unclassified   
345  Not applicable, International Standard Classif...           Unclassified   
346  Not applicable, International Standard Classif...           Unclassified   

             Gender Status of student in Canada  VALUE  
0       Male gender           Canadian students  129.0  
1       Male gender      International students    3.0  
2     Female gender           Canadian students  243.0  
3     Female gender      International students    3.0  
4    Gender unknown           Canadian students    3.0  
..              ...                         ...    ...  
342     Male gender           Canadian students    0.0  
343     Male gender           Canadian students  720.0  
344   Female gender           Canadian students  402.0  
345   Female gender      International students    0.0  
346  Gender unknown           Canadian students    3.0  

[347 rows x 9 columns]

### Question 5: Compute the number of female and number of male students enrolled at Memorial University of Newfoundland

We create a new dataframe for our mun students

In [6]:
df2 = df[df['Institution type'].str.contains('University') & df['GEO'].str.contains('Newfoundland and Labrador')]
df2

REF_DATE                        GEO Institution type  \
0    2018/2019  Newfoundland and Labrador       University   
1    2018/2019  Newfoundland and Labrador       University   
2    2018/2019  Newfoundland and Labrador       University   
3    2018/2019  Newfoundland and Labrador       University   
4    2018/2019  Newfoundland and Labrador       University   
..         ...                        ...              ...   
234  2018/2019  Newfoundland and Labrador       University   
235  2018/2019  Newfoundland and Labrador       University   
236  2018/2019  Newfoundland and Labrador       University   
237  2018/2019  Newfoundland and Labrador       University   
238  2018/2019  Newfoundland and Labrador       University   

    Registration status  \
0     Full-time student   
1     Full-time student   
2     Full-time student   
3     Full-time student   
4     Full-time student   
..                  ...   
234   Part-time student   
235   Part-time student   
236   Part-time student   
237   Part-time student   
238   Part-time student   

    International Standard Classification of Education (ISCED)  \
0                             Bachelor's or equivalent           
1                             Bachelor's or equivalent           
2                             Bachelor's or equivalent           
3                             Bachelor's or equivalent           
4                             Bachelor's or equivalent           
..                                                 ...           
234                             Doctoral or equivalent           
235                             Doctoral or equivalent           
236                             Doctoral or equivalent           
237                             Doctoral or equivalent           
238                             Doctoral or equivalent           

                                        Field of study          Gender  \
0                                        Education [1]     Male gender   
1                                        Education [1]     Male gender   
2                                        Education [1]   Female gender   
3                                        Education [1]   Female gender   
4                                        Education [1]  Gender unknown   
..                                                 ...             ...   
234  Agriculture, natural resources and conservatio...   Female gender   
235                     Health and related fields [10]     Male gender   
236                     Health and related fields [10]   Female gender   
237                                        Other  [12]     Male gender   
238                                        Other  [12]   Female gender   

    Status of student in Canada  VALUE  
0             Canadian students  129.0  
1        International students    3.0  
2             Canadian students  243.0  
3        International students    3.0  
4             Canadian students    3.0  
..                          ...    ...  
234      International students    3.0  
235           Canadian students   18.0  
236           Canadian students   42.0  
237           Canadian students    3.0  
238           Canadian students    3.0  

[239 rows x 9 columns]

Now we just need the value counts for each male and female on our new data frame.

In [7]:
df2['Gender'].value_counts()

Male gender       113
Female gender     108
Gender unknown     18
Name: Gender, dtype: int64